crates/host/src/lib.rs

```diff
@@ -70,7 +70,7 @@ impl AsRef<str> for ResourceRef<'_> {
         match self {
             // Resource ref must have originated from a URL, which can only be constructed from a
             // valid string
-            ResourceRef::File(path) => path.to_str().expect("invalid file reference URL"),
+            ResourceRef::File(url, _) => url,
```

Comment:
I think we can get away without needing to add an extra field. Because we have parsed this from a URL, we can always just add back on the `file://` to the string value returned back here when we use it. To solve the bug, when the event is sent, you can match on the enum type and preface if needed. The other option is to parse and trim the `file://` prefix on the wash side!